### PR TITLE
Made behavior to conform to ECMA for 'tail.call, pop, ret' case

### PR DESCRIPTION
**Pull request by rartemev**
Changes for #9880. Made runtime be conform to ECMA.

### CODE REVIEW

**Comment by rartemev** [comment]
@dotnet/jit-contrib @RussKeldorph please review

File: tests/issues.targets [hunk 1]
```diff
@@ -204,9 +204,6 @@
         <ExcludeList Include="$(XunitTestBinBase)\GC\LargeMemory\API\gc\suppressfinalize\suppressfinalize.cmd">
             <Issue>3392, test is useful to have because it can be run manually when making changes to the GC that can have effects in OOM scenarios, but not appropriate to run on our current test infrastructure.</Issue>
         </ExcludeList>
-        <ExcludeList Include="$(XunitTestBinBase)\JIT\Methodical\tailcall_v4\smallFrame\smallFrame.cmd">
```
Comment (sdmaclea) [inline tests/issues.targets]: Doesn't this test also need to be modified or removed so that it does not fail on all targets.

Comment (rartemev) [inline tests/issues.targets]: Actually, if we want to be conform to ECMA this test must be failed on all targets.

Comment (sdmaclea) [inline tests/issues.targets]: @rartemev Exactly my point.  I do not see how this informs CI that the failure is required.  And to pass regression if the test fails, but fail regression if the test passes.

Comment (rartemev) [inline tests/issues.targets]: I apologize, it is my mistake. Thank you

Comment (sdmaclea) [inline tests/issues.targets]: @rartemev No apology necessary.  I also must rely on others eyes to see what I missed.  Tests strangely seem to be passing...

Comment (rartemev) [inline tests/issues.targets]: Fixed

**Comment by rartemev** [comment]
@dotnet-bot test Windows_NT x64 Release Priority 1 Build and Test please

File: src/jit/flowgraph.cpp [hunk 2]
```diff
@@ -5455,15 +5455,15 @@ unsigned Compiler::fgMakeBasicBlocks(const BYTE* codeAddr, IL_OFFSET codeSize, B
                     if (!impIsTailCallILPattern(tailCall, opcode, codeAddr + sz, codeEndp, isRecursive,
                                                 &isCallPopAndRet))
                     {
-#ifdef _TARGET_AMD64_
+#ifndef FEATURE_CORECLR
```
Comment (RussKeldorph) [inline src/jit/flowgraph.cpp]: You need to retain the _TARGET_AMD64_ condition as well.  Even on the (full) .NET Framework, we wouldn't want to infect other architectures with this behavior.

Comment (rartemev) [inline src/jit/flowgraph.cpp]: Fixed

File: src/jit/importer.cpp [hunk 3]
```diff
@@ -15455,14 +15455,14 @@ bool Compiler::impReturnInstruction(BasicBlock* block, int prefixFlags, OPCODE&
     // We must have imported a tailcall and jumped to RET
     if (prefixFlags & PREFIX_TAILCALL)
     {
-#ifndef _TARGET_AMD64_
+#if FEATURE_CORECLR
```
Comment (RussKeldorph) [inline src/jit/importer.cpp]: defined(FEATURE_CORECLR) || !defined(_TARGET_AMD64_)?

Comment (rartemev) [inline src/jit/importer.cpp]: Fixed

File: src/jit/importer.cpp [hunk 4]
```diff
@@ -15455,14 +15455,14 @@ bool Compiler::impReturnInstruction(BasicBlock* block, int prefixFlags, OPCODE&
...
-#ifndef _TARGET_AMD64_
+#if defined(FEATURE_CORECLR) || !defined(_TARGET_AMD64_)
         // Jit64 compat:
         // This cannot be asserted on Amd64 since we permit the following IL pattern:
         //      tail.call
         //      pop
         //      ret
         assert(verCurrentState.esStackDepth == 0 && impOpcodeIsCallOpcode(opcode));
-#endif
+#endif // FEATURE_CORECLR && !_TARGET_AMD64_
```
Comment (sdmaclea) [inline src/jit/importer.cpp]: Comment does not match

**Pull request merged**